Source/DafnyLanguageServer/Workspace/TextDocumentLoader.cs

```diff
@@ -99,17 +97,41 @@ VerifierOptions verifierOptions
       }
 
       var verificationTasks = await verifier.GetVerificationTasksAsync(loaded, cancellationToken);
+      var initialViews = new ConcurrentDictionary<ImplementationId, ImplementationView>();
 
-      var initialViews = new Dictionary<ImplementationId, ImplementationView>();
       foreach (var task in verificationTasks) {
-        var status = await StatusFromImplementationTaskAsync(task);
-        var view = new ImplementationView(task.Implementation.tok.GetLspRange(), status, Array.Empty<Diagnostic>());
-        initialViews.Add(GetImplementationId(task.Implementation), view);
+        var status = StatusFromBoogieStatus(task.CacheStatus);
+        var id = GetImplementationId(task.Implementation);
+        if (task.CacheStatus is Completed completed) {
+          var view = new ImplementationView(task.Implementation.tok.GetLspRange(), status, GetDiagnosticsFromResult(loaded, completed.Result));
+          initialViews.TryAdd(GetImplementationId(task.Implementation), view);
+        } else if (loaded.ImplementationViewsView!.TryGetValue(id, out var existingView)) {
```

Comment:
Can you please add a comment to this method about how we ensure that `ImplementationViewsView` is not null? What is the method that needed to be called before, if any?